components/script/script_thread.rs

```diff
@@ -510,6 +511,9 @@ pub struct ScriptThread {
     /// A list of nodes with in-progress CSS transitions, which roots them for the duration
     /// of the transition.
     transitioning_nodes: DOMRefCell<Vec<JS<Node>>>,
+
+    /// Paint time metrics.
+    paint_time_metrics: Arc<PaintTimeMetrics>,
```

Comment:
It's not clear to me what this represents - ScriptThread is shared between an arbitrary number of similar-origin documents.